.github/workflows/publish.yml

```diff
@@ -11,21 +13,120 @@ permissions:
   contents: read
 
 jobs:
-  publish:
+  linux:
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: ubuntu-latest
+            target: x86_64
+          - runner: ubuntu-latest
+            target: x86
+          - runner: ubuntu-latest
+            target: aarch64
+          - runner: ubuntu-latest
+            target: armv7
+          - runner: ubuntu-latest
+            target: s390x
+          - runner: ubuntu-latest
+            target: ppc64le
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
```

Comment:
Looks like this is auto-generated, I assume this won't be an issue when used with other python versions that we support?

Reply:
My understanding was that since we delegate the build to `maturin`, any Python version which it supports should function basically the same way. I'm not sure if there is an affect on the built wheels though, in my head it should basically be Python source + compiled Rust but I will look into it further.

Reply:
Looked into this more, the ABI can change across Python versions, so we actually do we need to build wheels for every combo of OS/arch/Python that we want to support, but it looks like our GHA runners already have CPython 3.8-12 and a few versions of PyPy. Maturin auto-detects all of these and builds wheels for them (due to `--find-interpreter`)